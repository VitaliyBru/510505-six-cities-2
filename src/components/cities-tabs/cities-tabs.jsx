import React, {PureComponent} from "react";
import PropTypes from "prop-types";

export class CitiesTabs extends PureComponent {
  constructor(props) {
    super(props);

    this._handlerTabClick = this._handlerTabClick.bind(this);
  }

  render() {
    const {activeCity, citiesList} = this.props;

    return (
      <div className="tabs">
        <section className="locations container">
          <ul className="locations__list tabs__list">
            {
              citiesList.map((cityName) => {
                const elemMod = cityName === activeCity ? ` tabs__item--active` : ``;
                return (
                  <li className="locations__item" key={cityName}>
                    <a className={`locations__item-link tabs__item${elemMod}`} href="#" onClick={this._handlerTabClick}>
                      <span>{cityName}</span>
                    </a>
                  </li>
                );
              })
            }
          </ul>
        </section>
      </div>
    );
  }

  _handlerTabClick(evt) {
    const {onCitySelection, allCitiesOffers} = this.props;

    evt.preventDefault();
    onCitySelection(evt.currentTarget.firstChild.innerText, allCitiesOffers);
  }
}

CitiesTabs.propTypes = {
  activeCity: PropTypes.string.isRequired,
  citiesList: PropTypes.arrayOf(PropTypes.string).isRequired,
  allCitiesOffers: PropTypes.arrayOf(PropTypes.object).isRequired,
  onCitySelection: PropTypes.func.isRequired,
};
