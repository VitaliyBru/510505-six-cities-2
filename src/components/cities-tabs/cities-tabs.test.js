import React from "react";
import renderer from "react-test-renderer";
import {CitiesTabs} from "./cities-tabs.jsx";

it(`CitiesTabs renderer OK`, () => {
  const tree = renderer.create(
      <CitiesTabs
        activeCity={` `}
        citiesList={[]}
        allCitiesOffers={[]}
        onCitySelection={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
