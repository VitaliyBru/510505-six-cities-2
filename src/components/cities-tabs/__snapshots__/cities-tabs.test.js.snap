// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesTabs renderer OK 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    />
  </section>
</div>
`;
